src/clj/chronograph/db/acl.clj

```diff
@@ -4,30 +4,21 @@
             [chronograph.utils.time :as time])
   (:import (org.postgresql.util PGobject)))
 
-(defn create!
-  ([acl]
-   (create! db/datasource acl))
-  ([tx {:keys [user-id organization-id role]}]
-   (let [now (time/now)]
-     (sql/insert! tx
-                  :acls
-                  {:user-id user-id
-                   :organization-id organization-id
-                   :role (doto (PGobject.)
-                           (.setType "user_role")
-                           (.setValue role))
-                   :created-at now
-                   :updated-at now}
-                  db/sql-opts))))
+(defn create! [tx {:keys [user-id organization-id role]}]
+  (let [now (time/now)]
+    (sql/insert! tx
+                 :acls
+                 {:user-id user-id
+                  :organization-id organization-id
+                  :role (doto (PGobject.)
+                          (.setType "user_role")
+                          (.setValue role))
+                  :created-at now
+                  :updated-at now}
+                 db/sql-opts)))
 
-(defn find-acl
-  ([user-id organization-id]
-   (find-acl db/datasource user-id organization-id))
-  ([tx user-id organization-id]
-   (first
-    (sql/query tx
-               ["SELECT user_id, organization_id, role
-                   FROM acls
-                   WHERE user_id = ?
-                   AND organization_id = ?" user-id organization-id]
-               db/sql-opts))))
+(defn where [tx options]
+  (sql/find-by-keys tx :acls options db/sql-opts))
+
+(defn find-one [tx options]
```

Comment:
IMO, `find-one` should constrain options such that only one item is ever returned. Right now, if the caller makes a mistake, they may get back the first of a collection.

Preferably `where` and `find-one` would each execute specific queries that guarantee the correct return semantics (collection or nothing, item or nothing).

Reply:
Let's add a `LIMIT 1` clause so that many rows are not returned unnecessarily.